src/features/governance/hooks/useVote.ts

```diff
@@ -92,27 +93,27 @@ export const useVote = () => {
   );
 
   const getProposalVote = useCallback(
-    (proposalId: string) => {
-      const nodeData = votes[proposalId];
-      const localData = readFromCache(getVoteCacheKey(proposalId));
+    async (proposalId: string, address: string) => {
+      const nodeData = await getVotesForProposal(proposalId);
```

Comment:
thoughts: since this takes around 3-4 seconds to get the data, either we could optimise more or just prioritise the local node data

Reply:
maybe optimistic fetching